File: problems/problems_2553/solution.ts
function separateDigits(nums: number[]): number[] {
    const ans: Array<number> = [];
	for (let i: number = 0; i < nums.length; i++) {
		let num: number = nums[i];
		const cur: Array<number> = [];
		while (num > 0) {
			cur.push(num % 10);
			num = Math.floor(num / 10);
		}
		for (let j: number = cur.length - 1; j >= 0; j--) {
			ans.push(cur[j]);
		}
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return separateDigits(nums);
}

File: problems/problems_2553/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn separate_digits(nums: Vec<i32>) -> Vec<i32> {
		let mut ans: Vec<i32> = Vec::new();
		for mut num in nums {
			let mut cur: Vec<i32> = Vec::new();
			while num > 0 {
				cur.push(num % 10);
				num = num / 10;
			}
			cur.reverse();
			ans.append(&mut cur);
		}
		ans
    }
}

#[cfg(feature = "solution_2553")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let nums: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	json!(Solution::separate_digits(nums))
}
